### ISSUE TITLE
option_if_let_else shouldn't trigger when unwrap_or triggers

### ISSUE THREAD

**Issue by Hasnabruzzn** [rust-lang/rust-clippy#17071]
### Description

```
#[allow(clippy::disallowed_names)]
#[warn(clippy::option_if_let_else)]
fn main() {
    let optional = Some(42);
    let _ = if let Some(foo) = optional {
        foo
    } else {
        5
    };
}
```

The output is unnecessarily verbose, as it triggers two separate lints simultaneously (clippy::manual_unwrap_or and clippy::option_if_let_else)

This is from the crafting errors workshop at RustWeek by @jdonszelmann and @estebank. The first part of the task was #5821 , which is already closed.

IMHO, unwrap_or (when it applies) would be always preferable over map_or. Or are there any other cases? (If not I'd be almost finished)

### Version

```text

```

**Comment by estebank** [comment]
> IMHO, unwrap_or (when it applies) would be always preferable over map_or.

I would tend to agree.
